scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+      }
+    }
+    val dependency = MavenDependency.newBuilder()
+      .setArtifactName(mavenDependency.artifactName)
+      .setArtifactVersion(mavenDependency.artifactVersion)
+      .addAllLicense(updateLicenseList)
+      .setOriginOfLicense(OriginOfLicenses.UNKNOWN)
+
+    if (updateLicenseList.isNotEmpty()) {
+      if (numberOfLicensesToBeProvidedManually == updateLicenseList.size) {
+        dependency.originOfLicense = OriginOfLicenses.MANUAL
+      } else if (numberOfLicensesToBeProvidedManually == 0) {
+        dependency.originOfLicense = OriginOfLicenses.ENTIRELY_FROM_POM
+      } else if (numberOfLicensesToBeProvidedManually != updateLicenseList.size) {
+        dependency.originOfLicense = OriginOfLicenses.PARTIALLY_FROM_POM
+      }
+    }
+    finalUpdatedList.add(dependency.build())
+  }
+  return finalUpdatedList
+}
+
+/** Retrieves the list of [MavenDependency] from maven_dependencies.textproto. */
+private fun retrieveMavenDependencyList(): List<MavenDependency> {
+  return getProto(
+    "maven_dependencies.pb",
+    MavenDependencyList.getDefaultInstance()
+  ).mavenDependencyList.toList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun getProto(
+  textProtoFileName: String,
+  proto: MavenDependencyList
+): MavenDependencyList {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+  val protoObject = FileInputStream(protoBinaryFile).use {
+    builder.mergeFrom(it)
+  }.build() as MavenDependencyList
+  return protoObject
+}
+
+fun parseArtifactName(artifactName: String): String {
+  var colonIndex = artifactName.length - 1
+  while (artifactName.isNotEmpty() && artifactName[colonIndex] != ':') {
+    colonIndex--
+  }
+  val artifactNameWithoutVersion = artifactName.substring(0, colonIndex)
+  val parsedArtifactNameBuilder = StringBuilder()
+  for (index in artifactNameWithoutVersion.indices) {
+    if (artifactNameWithoutVersion[index] == '.' || artifactNameWithoutVersion[index] == ':' ||
+      artifactNameWithoutVersion[index] == '-'
+    ) {
+      parsedArtifactNameBuilder.append(
+        '_'
+      )
+    } else {
+      parsedArtifactNameBuilder.append(artifactNameWithoutVersion[index])
+    }
+  }
+  return parsedArtifactNameBuilder.toString()
+}
+
+fun runBazelQueryCommand(rootPath: String): List<String> {
+  val rootDirectory = File(rootPath).absoluteFile
+  val bazelClient = BazelClient(rootDirectory)
+  val bazelQueryDepsNames = mutableListOf<String>()
+  val output = bazelClient.executeBazelCommand(
+    "query",
+    "\'deps(deps(//:oppia)",
+    "intersect",
+    "//third_party/...)",
+    "intersect",
+    "@maven//...\'"
+  )
+  output.forEach { dep ->
+    bazelQueryDepsNames.add(dep.substring(9, dep.length))
```

Comment:
Why the substring?

Reply:
To omit `@maven//:`.

Reply:
Replaced with `dep.substring(MAVEN_PREFIX.length, dep.length)`